prow/metrics/metrics.go

```diff
@@ -53,9 +54,15 @@ func ExposeMetrics(component string, pushGateway config.PushGateway) {
 func serveMetrics(reg *prometheus.Registry) {
 	var handler http.Handler
 	if reg == nil {
-		handler = promhttp.Handler()
+		handler = promhttp.HandlerFor(
+			prometheus.Gatherers{prometheus.DefaultGatherer, ctrlruntimemetrics.Registry},
+			promhttp.HandlerOpts{},
+		)
 	} else {
-		handler = promhttp.HandlerFor(reg, promhttp.HandlerOpts{})
+		handler = promhttp.HandlerFor(
+			prometheus.Gatherers{reg, ctrlruntimemetrics.Registry},
+			promhttp.HandlerOpts{},
+		)
```

Comment:
nit: I think this would be less verbose and therefore better readable:

```
var gatherer ..
if reg == nil {
  gatherer = p.DefaultGatherer
} else {
  gatherer = reg
}
handler := ...Gatherers{gatherer, ...
```

Maybe even change `serveMetrics` to accept a `prometheus.Gatherer` right away?

/hold
Fix or unhold, your choice.

Reply:
Good point, fixed